File: palindrome.py
import re

def is_palindrome(s):
    # Normalize the input string

    s = s.lower()  # Convert to lowercase
    s = re.sub(r'[^a-z0-9]', '', s)  # Remove non-alphanumeric characters
    if s == s[::-1]:# Check if the normalized string is equal to its reverse
        print("Palindrome")
    else:
        print("Not Palindrome")

# Get input from user
s = input("Enter a string: ")
is_palindrome(s)

File: palindrome.cpp
#include <iostream>
#include <algorithm>
#include <cctype>
#include <string>
using namespace std;

bool is_palindrome(string s) {
    // Normalize the input string
    transform(s.begin(), s.end(), s.begin(), ::tolower); // Convert to lowercase
    s.erase(remove_if(s.begin(), s.end(), [](char c) { return !isalnum(c); }), s.end()); // Remove non-alphanumeric characters

    // Check if the normalized string is equal to its reverse
    string rev = s;
    reverse(rev.begin(), rev.end());
    return s == rev;
}

int main() {
    string s;
    cout << "Enter a string: ";
    getline(cin, s);

    if (is_palindrome(s)) {
        cout << "Palindrome" << endl;
    } else {
        cout << "Not Palindrome" << endl;
    }

    return 0;
}
